### PR TITLE
Cb 418

### PR DESCRIPTION
**Pull request by adamconder**

### CODE REVIEW

**Comment by chris-ianson** [comment]
@adamconder @adamfenwick @anujakhare Ready for code review

File: app/uk/gov/hmrc/cb/controllers/CBRoutes.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+package uk.gov.hmrc.cb.controllers
+
+/**
+  * Created by chrisianson on 24/06/16.
+  */
+trait CBRoutes {
+  val initialController = uk.gov.hmrc.cb.controllers.routes.UpdateChildBenefitController.get()
+  val technicalDifficulties = uk.gov.hmrc.cb.controllers.routes.TechnicalDifficultiesController.get()
+  //val initialController = "/child-benefit/update-child-benefit"
```
Comment (adamconder) [inline app/uk/gov/hmrc/cb/controllers/CBRoutes.scala]: @chris-ianson Remove comment

File: app/uk/gov/hmrc/cb/controllers/child/ChildBirthCertificateReferenceController.scala [hunk 2]
```diff
@@ -59,45 +60,57 @@ trait ChildBirthCertificateReferenceController extends ChildBenefitController {
...
-              if (child.hasBirthCertificateReferenceNumber) {
-                Logger.debug(s"[ChildBirthCertificateReferenceController][get] child does exist at index")
-                val model : ChildBirthCertificateReferencePageModel = child
-                view(Ok, form.fill(model), id)
-              } else {
-                resultWithNoChild
+      val resultWithNoChild = view(Ok, form, id)
+      cacheClient.loadPayload().map {
+        payload =>
+          Logger.error(s"[ChildBirthCertificateReferenceController][get] loaded payload")
```
Comment (adamconder) [inline app/uk/gov/hmrc/cb/controllers/child/ChildBirthCertificateReferenceController.scala]: @chris-ianson This should be debug not error

File: app/uk/gov/hmrc/cb/controllers/child/ChildBirthCertificateReferenceController.scala [hunk 3]
```diff
@@ -59,45 +60,57 @@ trait ChildBirthCertificateReferenceController extends ChildBenefitController {
...
+                case None =>
+                  val children = addChild(id, model, List())
+                  Payload(children = children)
               }
+
+              saveToKeystore(modifiedPayload, id)
           } recover {
             case e : Exception =>
-              Logger.error(s"[ChildBirthCertificateReferenceController][post] keystore exception whilst loading children: ${e.getMessage}}")
+              Logger.error(s"[ChildBirthCertificateReferenceController][get] keystore exception whilst loading children: ${e.getMessage}")
```
Comment (adamconder) [inline app/uk/gov/hmrc/cb/controllers/child/ChildBirthCertificateReferenceController.scala]: @chris-ianson This should be post not get

File: app/uk/gov/hmrc/cb/controllers/child/ChildDateOfBirthController.scala [hunk 4]
```diff
@@ -57,22 +58,29 @@ trait ChildDateOfBirthController extends ChildBenefitController {
...
-            child =>
-              if (child.hasDateOfBirth) {
-                val model : ChildDateOfBirthPageModel = child
-                view(Ok, form.fill(model), id)
-              } else {
-                resultWithNoChild
+      val resultWithNoChild = view(Ok, form, id)
+      cacheClient.loadPayload().map {
+        payload =>
+          Logger.error(s"[ChildDateOfBirthController][get] loaded payload")
```
Comment (adamconder) [inline app/uk/gov/hmrc/cb/controllers/child/ChildDateOfBirthController.scala]: @chris-ianson this should not be error.

File: app/uk/gov/hmrc/cb/controllers/claimant/ClaimantNameController.scala [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  }
+
+  private def redirectConfirmation() = Redirect(uk.gov.hmrc.cb.controllers.routes.SubmissionConfirmationController.get())
+
+  def get() = CBSessionProvider.withSession {
+    implicit request =>
+      val resultWithNoClaimant = view(Ok, form)
+      cacheClient.loadPayload().map {
+        payload =>
+          Logger.error(s"[ClaimantNameController][get] loaded payload")
```
Comment (adamconder) [inline app/uk/gov/hmrc/cb/controllers/claimant/ClaimantNameController.scala]: @chris-ianson This should not be error

File: app/uk/gov/hmrc/cb/controllers/claimant/ClaimantNameController.scala [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+          payload.fold(
+            resultWithNoClaimant
+          )(
+            cache => {
+              cache.claimant match {
+                case Some(x) =>
+                  Logger.error(s"[ClaimantNameController][get] loaded claimant")
```
Comment (adamconder) [inline app/uk/gov/hmrc/cb/controllers/claimant/ClaimantNameController.scala]: @chris-ianson this should not be error.

**Pull request merged**